runtime/src/main/scala/akka/grpc/internal/NettyClientUtils.scala

```diff
@@ -25,21 +27,28 @@ object NettyClientUtils {
    * INTERNAL API
    */
   @InternalApi
-  def createChannel(settings: GrpcClientSettings): ManagedChannel = {
-    var builder =
-      NettyChannelBuilder
-        .forAddress(settings.host, settings.port)
-        .flowControlWindow(NettyChannelBuilder.DEFAULT_FLOW_CONTROL_WINDOW)
-
-    builder = settings.trustedCaCertificate.map(c => GrpcSslContexts.forClient.trustManager(loadCert(c)).build)
-      .map(ctx => builder.negotiationType(NegotiationType.TLS).sslContext(ctx))
-      .getOrElse(builder.negotiationType(NegotiationType.PLAINTEXT))
-
-    builder = settings.overrideAuthority.map(builder.overrideAuthority(_)).getOrElse(builder)
-
-    builder = settings.userAgent.map(builder.userAgent(_)).getOrElse(builder)
-
-    builder.build
+  def createChannel(settings: GrpcClientSettings)(implicit ec: ExecutionContext): Future[ManagedChannel] = {
+    settings.serviceDiscovery.lookup(settings.name, settings.resolveTimeout).map { targets: SimpleServiceDiscovery.Resolved =>
+      if (targets.addresses.nonEmpty) {
+        val target = targets.addresses(ThreadLocalRandom.current().nextInt(targets.addresses.size))
+        var builder =
+          NettyChannelBuilder
+            .forAddress(target.host, target.port.getOrElse(settings.defaultPort))
+            .flowControlWindow(NettyChannelBuilder.DEFAULT_FLOW_CONTROL_WINDOW)
+
+        builder = settings.trustedCaCertificate.map(c => GrpcSslContexts.forClient.trustManager(loadCert(c)).build)
+          .map(ctx => builder.negotiationType(NegotiationType.TLS).sslContext(ctx))
+          .getOrElse(builder.negotiationType(NegotiationType.PLAINTEXT))
+
+        builder = settings.overrideAuthority.map(builder.overrideAuthority(_)).getOrElse(builder)
+
+        builder = settings.userAgent.map(builder.userAgent(_)).getOrElse(builder)
+
+        builder.build
+      } else {
+        throw new IllegalStateException("No targets returned for name: " + settings.name)
```

Comment:
Not super nice to throw... we could also replace `map` above with `flatMap` and return a failed future here?